### PR TITLE
Introduce `subTypes` to `BannerComponents`

### PR DESCRIPTION
**Pull request by abhishek1508**

### CODE REVIEW

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java [hunk 1]
```diff
@@ -74,21 +74,68 @@ public abstract class BannerComponents extends DirectionsJsonObject
...
+  public static final String EXPRESSWAY_EXIT = "exit";
+
   /**
-   * This view gives guidance through signboards and is used to complete maneuvers.
+   * Junction View. Bird’s-eye artist’s rendition view of the overhead signage and
+   * preferred road lane arrow on motorways where the road bifurcates into 2 or
+   * more motorway trunk roads.
    */
-  public static final String SIGNBOARD = "signboard";
+  public static final String JCT = "jct";
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java:110]: This should be undocumented.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java [hunk 2]
```diff
@@ -74,21 +74,68 @@ public abstract class BannerComponents extends DirectionsJsonObject
...
    */
   @SuppressWarnings("checkstyle:javadocvariable")
   public static final String SAPAGUIDEMAP = "sapaguidemap";
 
   /**
-   * This view gives guidance through junctions and is used to complete maneuvers.
+   * Advanced 2D signboard. These are vendor enhanced detailed signboards.
    */
-  public static final String JCT = "jct";
+  public static final String SIGNBOARD = "signboard";
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java:131]: Same as above.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java [hunk 3]
```diff
@@ -178,7 +231,7 @@ public static Builder builder() {
    * @return String type from above list
    */
   @Nullable
-  @BannerComponentsType
+  @BannerComponentsSubType
   public abstract String subType();
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java:235]: Same as above.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java [hunk 4]
```diff
@@ -74,21 +74,68 @@ public abstract class BannerComponents extends DirectionsJsonObject
    */
   public static final String GUIDANCE_VIEW = "guidance-view";
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java:75]: Same as above.

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/BannerComponentTest.java [hunk 5]
```diff
@@ -216,6 +216,66 @@ public void testToFromJsonLaneText() {
     assertEquals(bannerComponents, bannerComponentsFromJson);
   }
 
+  @Test
+  public void testToFromJsonAfterTollSubType() {
+
+    BannerComponents bannerComponents = BannerComponents.builder()
+        .text("")
+        .type("guidance-view")
+        .subType("aftertoll")
```
Comment (LukasPaczos) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/BannerComponentTest.java:225]: Let's use the constants instead.

**Comment by abhishek1508** [comment]
@LukasPaczos Addressed all the comments above!

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java [hunk 6]
```diff
@@ -72,23 +72,73 @@ public abstract class BannerComponents extends DirectionsJsonObject
   /**
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java:72]:
```suggestion
  /*
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java [hunk 7]
```diff
@@ -72,23 +72,73 @@ public abstract class BannerComponents extends DirectionsJsonObject
...
+  public static final String EXPRESSWAY_ENTRANCE = "entrance";
+
+  /*
+   * Expressway Exit. Bird’s-eye artist’s rendition view of the overhead signage and
+   * preferred road lane arrow at an exit ramp from an expressway.
+   */
+  @SuppressWarnings("checkstyle:javadocvariable")
+  public static final String EXPRESSWAY_EXIT = "exit";
+
   /**
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java:106]:
```suggestion
  /*
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java [hunk 8]
```diff
@@ -72,23 +72,73 @@ public abstract class BannerComponents extends DirectionsJsonObject
...
   /*
-   * This view gives guidance through guide maps and is used to complete maneuvers.
+   * SAPA Guide Map. Vertical artist’s rendition guide map of an SAPA rest area
+   * showing various facilities icons such as restaurants, restrooms and parking areas.
+   * The scale is approx. 1:5K.
    */
   @SuppressWarnings("checkstyle:javadocvariable")
   public static final String SAPAGUIDEMAP = "sapaguidemap";
 
   /**
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java:130]:
```suggestion
  /*
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java (hunk 3 above)
Comment (LukasPaczos) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java:239]: Still needs the javadoc to be removed.

**Review by LukasPaczos — APPROVED** [review]

**Pull request merged**